project/src/main/ui/chat/chat-bool-parser.gd

```diff
@@ -0,0 +1,291 @@
+class_name ChatBoolParser
+"""
+Parses a string boolean statement into a tree of ChatExpressions.
```

Comment:
Change `boolean statement` to `boolean expression` for consistency

Reply:
"Expression" is a little overloaded, as the tree is considered an 'expression tree'. I think statement might be less ambiguous.

Reply:
After a heated discussion with myself, I've settled upon the phrase `string boolean expression`